pkg/controller/common.go

```diff
@@ -484,26 +484,40 @@ func (c *SharedState) addPod(pod *v1.Pod, updateOption updateWithLock) {
 	klog.V(5).Infof("Pod spec of pod %s is: %+v. With volumes: %+v", pod.Name, pod.Spec, pod.Spec.Volumes)
 
 	// If the claims already exist for the podKey, add them to a set
-	value, _ := c.podToClaimsMap.LoadOrStore(podKey, []string{})
+	value, _ := azr.getSharedState().podToClaimsMap.LoadOrStore(podKey, []string{})
 	claims := value.([]string)
 	claimSet := set{}
 	for _, claim := range claims {
 		claimSet.add(claim)
 	}
 
 	for _, volume := range pod.Spec.Volumes {
+		if volume.AzureDisk != nil {
```

Comment:
Please add a comment that this is to handle CSI Migration from the in-tree plug-in. Should we only do this if the CSI Migration feature gate is enabled?